backend/tests/Feature/PublicationTest.php

```diff
@@ -178,388 +171,151 @@ public function testQueryFiltersHiddenItems()
     }
 
     /**
-     * @dataProvider deletePublicationUserViaMutationAsAnEditorProvider
+     * @dataProvider allSubmissionRoles
+     * @param string $role
      * @return void
      */
-    public function testDeletePublicationUserViaMutationAsAnEditor($userRoleId, bool $allowed)
+    public function testApplicationAdminCanAssignAnyRole(string $role): void
     {
-        /** @var User $editor */
-        $editor = User::factory()->create();
-        $editor->assignRole(Role::EDITOR);
-        $this->actingAs($editor);
-        $user_to_be_deleted = User::factory()->create();
-        $publication_user_role_id_is_invalid = intval($userRoleId) <= 0;
-        $publication = Publication::factory()
-            ->hasAttached(
-                $user_to_be_deleted,
-                [
-                    'role_id' => $publication_user_role_id_is_invalid ? Role::EDITOR_ROLE_ID : $userRoleId, //TODO: What is the point of this ? If the role is invalid we just pretend it says EDITOR?
-                ]
-            )
-            ->hasAttached(
-                $editor,
-                [
-                    'role_id' => 3,
-                ]
-            )
-            ->create([
-                'name' => 'Test Publication for Publication User Unassignment Via Mutation',
-            ]);
-
-        $publication_user = PublicationUser::firstOrFail();
-        $response = $this->graphQL(
-            'mutation DeletePublicationUser ($role_id: ID!, $publication_id: ID!, $user_id: ID!) {
-                deletePublicationUser(
-                    publication_user: { role_id: $role_id, publication_id: $publication_id, user_id: $user_id }
-                ) {
-                    id
-                }
-            }',
-            [
-                'role_id' => $userRoleId,
-                'publication_id' => $publication->id,
-                'user_id' => $user_to_be_deleted->id,
-            ]
-        );
-        $expected_mutation_response = null;
-        if ($allowed) {
-            $expected_mutation_response = [
-                'deletePublicationUser' => [
-                    'id' => (string)$publication_user->id,
-                ],
-            ];
-        }
-        $response->assertJsonPath('data', $expected_mutation_response);
-    }
+        $this->beAppAdmin();
 
-    /**
-     * @return array
-     */
-    public function createPublicationUserViaMutationAsAnApplicationAdministratorProvider(): array
-    {
-        //@codingStandardsIgnoreStart
-        return [
-            //User Role                         Allowed?
-            [ Role::SUBMITTER_ROLE_ID,          false ],
-            [ Role::REVIEWER_ROLE_ID,           false ],
-            [ Role::REVIEW_COORDINATOR_ROLE_ID, false ],
-            [ Role::EDITOR_ROLE_ID,             true  ],
-            [ 0,                                false ],
-            [ '',                               false ],
-            [ null,                             false ],
```

Comment:
We don't need to test roles values anymore because the client doesn't supply them anymore.  They're determined by which of the relationships are being updated ($publication->publication_admins(), $publication->editors()).  This is one of the primary advantages to this refactor, the role numbers aren't part of something the front end needs to be concerned about anymore.

Reply:
That makes sense.